chalice/awsclient.py

```diff
@@ -140,8 +198,12 @@ def update_function(self,
         the targeted lambda function.
         """
         lambda_client = self._client('lambda')
-        return_value = lambda_client.update_function_code(
-            FunctionName=function_name, ZipFile=zip_contents)
+        try:
+            return_value = lambda_client.update_function_code(
+                FunctionName=function_name, ZipFile=zip_contents)
+        except Exception as e:
```

Comment:
Same thing, we should really only have `except Exceptions` and the top most layer of the code.  Restrict this to the subset of exceptions we care about.